### PR TITLE
Preview values in collapsed state in JsonDisplay

### PR DESCRIPTION
**Pull request by fiskus**
JSON file before:
Screenshot from 2023-01-11 13-54-11
Screenshot from 2023-01-11 13-54-22

JSON file after:
Screenshot from 2023-01-11 13-54-30
Screenshot from 2023-01-11 13-54-42

Package meta before:
Screenshot from 2023-01-11 13-55-53
Screenshot from 2023-01-11 13-56-14

Package meta after:
Screenshot from 2023-01-11 13-55-59
Screenshot from 2023-01-11 13-56-17

- [x] Unit tests
- [x] [Changelog](CHANGELOG.md) entry (skip if change is not significant to end users, e.g. docs only)

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Entries inside each section should be ordered by type:
 * [Added] Add link to file from Athena results ([#3242](https://github.com/quiltdata/quilt/pull/3242))
 * [Added] Add link to package revisions from package list ([#3256](https://github.com/quiltdata/quilt/pull/3256))
 * [Changed] Make file preview wrapper consistently 100% width ([#3245](https://github.com/quiltdata/quilt/pull/3245))
+* [Changed] Show collapsed values in JSON viwer ([#3249](https://github.com/quiltdata/quilt/pull/3249))
```
Comment (nl0) [inline docs/CHANGELOG.md:22]:
```suggestion
* [Changed] Show collapsed values in JSON viewer ([#3249](https://github.com/quiltdata/quilt/pull/3249))
```

**Review by nl0 — DISMISSED** [review]
wow this is super useful

File: catalog/app/components/JsonDisplay/JsonDisplay.js [hunk 2]
```diff
@@ -282,13 +294,19 @@ function JsonDisplayInner(props) {
...
 }
 
-function useCurrentBreakpointWidth() {
-  const t = M.useTheme()
-  return ['sm', 'md', 'lg'].reduce(
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    (acc, b) => (M.useMediaQuery(t.breakpoints.up(b)) ? t.breakpoints.width(b) : acc),
-    320, // min supproted width
-  )
+function useCurrentBreakpointWidth(ref) {
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/components/JsonDisplay/JsonDisplay.js:297]: I'll rename it, but API for utils is not obvious to me (should the argument be Ref or DOM node), so I'd postpone moving it to utils

File: catalog/app/utils/JSONOneliner.spec.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      limit: 30,
+      result: `[ 1, <…1>, true, false, null ]`,
+    }
+    const d = JSONOneliner.print(t.value, t.limit, true)
+    expect(printData(d)).toBe(t.result)
+    expect(d.availableSpace).toBe(t.limit - t.result.length)
+  })
+
+  it('should show key only if value is too long and not enough space', () => {
+    const t = {
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/utils/JSONOneliner.spec.ts:105]: Yeah, it was overthinking :)

File: catalog/app/utils/JSONOneliner.spec.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+function printData(data: JSONOneliner.SyntaxData) {
+  return data.parts.reduce((memo, p) => memo + p.value, '')
+}
+
+describe('utils/JSONOneliner', () => {
+  it('should print empty brackets for empty array', () => {
+    const t = {
+      value: [],
+      limit: 10,
+      result: `[  ]`,
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/utils/JSONOneliner.spec.ts]: I think of two ways to do this:
1. data is untouched (two items: `[ ` and ` ]`), but we can remove extra spaces on post-processing
2. refactor data and introduce new item type: Space. So, Separator (` ,`) becomes Space + Separator (` ` + `,`), Brace (` ]`) becomes Space + Brace (` ` + ` ]`)

2 feels more right for me, but I'd postpone it for future refactoring

**Review by nl0 — APPROVED** [review]

**Pull request merged**